### PR TITLE
Fix hash implementation for `()`. [ECR-442]

### PR DESCRIPTION
**Pull request by SerhiiOryshych**

### CODE REVIEW

**Comment by DarkEld3r** [comment]
Please add a changelog entry.

File: exonum/CHANGELOG.md [hunk 1]
```diff
@@ -59,6 +59,9 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 - `storage::Error` implements `failure::Fail` instead of `std::error::Error`. (#474)
 
+- `CryptoHash` for `()` now correctly calculated hash instead
```
Comment (vitvakatu) [inline exonum/CHANGELOG.md]: calculates?

File: exonum/src/crypto.rs [hunk 2]
```diff
@@ -612,7 +612,7 @@ impl CryptoHash for i64 {
 
 impl CryptoHash for () {
     fn hash(&self) -> Hash {
-        Hash::zero()
+        hash(&[])
```
Comment (slowli) [inline exonum/src/crypto.rs]: I wonder if it could make sense to return a constant here. Is this doable without `const fn` support?

Comment (DarkEld3r) [reply exonum/src/crypto.rs]: I suppose it's not possible without things like `lazy_static`.

Comment (matklad) [reply exonum/src/crypto.rs]: >I suppose it's not possible without things like lazy_static.

I'd say that transmuting an `[u8; 32]` would be a nice solution

Comment (matklad) [reply exonum/src/crypto.rs]: Provided that we add a tests that checks that the the result of transmute is indeed equal to hash(&[]).

**Pull request merged**